Slightly hacky support for android - Updated

adamrainsby opened the pull request:
My last pull request had a bug. This one should work.

https://github.com/naoufal/react-native-accordion/issues/16

adamrainsby committed:
Hide and show accordion content for android

adamrainsby committed:
Add Platform

adamrainsby committed:
Add empty function to onLayout so views can be measured

gr4yscale commented:
Hi @adamrainsby, can you please explain what the thinking is behind these changes? I see you added the empty onLayout but not sure what the purpose is, is this still a work in progress or does it work with these changes?

adamrainsby commented:
@gr4yscale See this issue for the onLayout thing. https://github.com/facebook/react-native/issues/3282

adamrainsby commented:
If you want to contribute and get it working with the animation again that would be great. At the moment it literally just shows and hides the content without any animation so it definitely can be improved.

naoufal commented:
Hey guys, thanks for moving this forward.

I would opt for a solution that retains the animations.  @gr4yscale Have you had any luck fixing this?

adamrainsby committed:
rename index.ios.js to index.js

adamrainsby commented:
@naoufal Just to be clear, the animations still work on iOS but it just opens and closes on android. That is an improvement on it not working at all on android.

I think it should be merged in and then we can open an issue for animations on android?

naoufal commented:
@adamrainsby you're right. Before merging, could you post a video of the example running on Android?

adamrainsby commented:
@naoufal Yep
![accordion](https://cloud.githubusercontent.com/assets/7249067/14576689/8ed83e84-0363-11e6-9ceb-2db661ec33c9.gif)

adamrainsby committed:
Add android example

adamrainsby commented:
@naoufal I had to do a bit of work to get the example app running on android. I've updated my PR with those changes.

iOS specific components like NavigatorIOS were stripped out for the android version but they are still there when running on iOS. I also updated the version of react native to the latest version because there was a bug where the background would disappear after pressing the accordion.

Not sure the exact version this was fixed but for now we can say there is android support from 0.23?

![accordion](https://cloud.githubusercontent.com/assets/7249067/14577856/ffc6d642-0371-11e6-9f3b-62b7b7e36258.gif)

adamrainsby commented:
@naoufal Is this ok?

cfournel commented:
@naoufal if this is okay , could you please merge the code and do a release ? i can t use your great module at the moment since my app has to be android / ios compatible. Thanks you !

GeoffreyPlitt commented:
+1

Commit:
compatibility updates

naoufal commented:
@adamrainsby Sorry for being MIA all.  @adamrainsby If you rebase this branch, I'll gladly merge it.

cfournel commented:
@naoufal  : Here  is my test of @adamrainsby . Since i have copied directly @adamrainsby fork into my node_modules dir , i had to install react-tween-state and everything works fine ( except animations of course, but may that could be related to a tween effect or duration ? ) .. here is a screen cast : 

![](https://i.imgur.com/jXmeNmN.gif)

adamrainsby commented:
@huitiemesens You can put "adamrainsby/react-native-accordion" instead of a version number in the package.json and npm install as usual instead of copy and pasting.

cfournel commented:
@adamrainsby thanks for this tips, i m more a composer guy , i didn t know that could be done within a package file

parveen-wizni commented:
hi,
I am facing some issue. I am using "react-native-accordion": "0.2.4" library for my project. This is working fine in IOS. when I run my app in android device in debugging mode. Then app is running fine. But if  I used signed apk in same device then it is give this error. here I  paste my application crash report log which are shown in android monitor. Error given below.

FATAL EXCEPTION: mqt_native_modules
                                                 Process: com.ampsmobile, PID: 2139
                                                 com.facebook.react.modules.core.JavascriptException: Requiring unknown module "react-native-accordion", stack:
                                                 t@2:387
                                                 i@2:177
                                                 <unknown>@451:139
                                                 t@2:511
                                                 i@2:177
                                                 <unknown>@449:139
                                                 t@2:511
                                                 i@2:177
                                                 <unknown>@448:203
                                                 t@2:511
                                                 i@2:177
                                                 <unknown>@447:212
                                                 t@2:511
                                                 t@2:240
                                                 i@2:177
                                                 component@44:489
                                                 u@44:130
                                                 o@44:48
                                                 createScene@44:1781
                                                 _renderScene@244:15724
                                                 <unknown>@244:16158
                                                 render@244:16032
                                                 _renderValidatedComponentWithoutOwnerOrContext@152:5072
                                                 _renderValidatedComponent@152:5200
                                                 _updateRenderedComponent@152:4595
                                                 _performComponentUpdate@152:4395
                                                 updateComponent@152:3672
                                                 performUpdateIfNecessary@152:3193
                                                 performUpdateIfNecessary@132:667
                                                 r@130:571
                                                 perform@135:494
                                                 perform@135:494
                                                 perform@130:1539
                                                 P@130:1711
                                                 closeAll@135:1131
                                                 perform@135:581
                                                 batchedUpdates@129:456
                                                 s@130:761
                                                 a@149:69
                                                 enqueueSetState@149:917
                                                 setState@29:339
                                                 push@244:13537
                                                 onPressRow@429:2254
                                                 onPress@437:2706
                                                 touchableHandlePress@295:1171
                                                 _performSideEffectsForTransition@219:8188
                                                 _receiveSignal@219:6697
                                                 touchableHandleResponderRelease@219:4183
                                                 a@113:72
                                                 o@111:392
                                                 a@111:583
                                                 f@109:163
                                                 g@109:266
                                                 i@115:88
                                                 processEventQueue@109:1386
                                                 s@163:92
                                                 handleTopLevel@163:182
                                                 _receiveRootNodeIDEvent@162:617
                                                 receiveTouches@162:999
                                                 value@60:3582
                                                 <unknown>@60:1801
                                                 k@60:436
                                                 value@60:1773

cfournel commented:
In order to keep it clean @naoufal could you please merge this one ? Thanks

adamrainsby commented:
Apologies @huitiemesens, I still need to rebase. I'll try to do it soon.

cfournel commented:
@adamrainsby  don t worry i already switch for your fork in my package.json , since @naoufal seems afk for a while now ... 

In order to fix a great bug ( avoiding to compile a signed android apk ) could you please take a look at https://github.com/naoufal/react-native-accordion/pull/33/commits ( its basically renaming index.ios.js to index.js ) ! Thanks you !

adamrainsby commented:
@huitiemesens I already did that in my fork. Are you still having issues with it?

adamrainsby committed:
Update react native on example app

adamrainsby commented:
@naoufal Rebased, could you merge?
